src/deploy/kubernetes_functions.js

```diff
@@ -1,260 +1,292 @@
-        const {
-            namespace
-        } = params;
-        if ((!namespace)) {
-            throw new Error('In order to protect other namespaces, namespace param is mandatory');
+
+    async deploy_server({ image, server_yaml, envs, cpu, mem, pv }) {
+        const server_details = {};
+        try {
+            let resources_file_path = path.join(this.output_dir, `${this._namespace}.server_deployment.${Date.now()}.json`);
+            // modify resources and write to temp yaml
+            const resources = await this.read_resources(server_yaml);
+            const statefulset = resources.find(res => res.kind === 'StatefulSet');
+            const pod_name = `${statefulset.metadata.name}-0`;
+            this.update_statefulset({ statefulset, image, envs, cpu, mem, pv });
+            await this.write_resources(resources_file_path, resources);
+
+            console.log('deploying server resources from file', resources_file_path);
+            await this.kubectl(`apply -f ${resources_file_path}`);
+
+            // get services info
+            console.log('getting s3 and managements services address');
+            const { address: s3_addr, ports: s3_ports } = await this.get_service_address('s3');
+            const { address: mgmt_addr, ports: mgmt_ports } = await this.get_service_address('noobaa-mgmt', 'mgmt-https');
+            server_details.services = {
+                namespace: this._namespace,
+                s3: { address: s3_addr, ports: s3_ports },
+                mgmt: { address: mgmt_addr, ports: mgmt_ports },
+            };
+            server_details.pod_name = pod_name;
+
+            // wait for server pod to be ready
+            await this.wait_for_pod_ready(pod_name, () => this.test_http_req(`http://${mgmt_addr}:${mgmt_ports.mgmt}/version`, 200));
+
+            this.server_details = server_details;
+            return server_details;
+        } catch (err) {
+            console.error('failed to deploy server. error:', err);
+            throw err;
         }
-        const run_with_namespace = await this.run_with_namespace(namespace);
-        console.log(await promise_utils.exec(`kubectl delete pvc --all ${run_with_namespace}`, { return_stdout: true }));
     }
 
-    async delete_namespace(params) {
-        const {
-            namespace
-        } = params;
-        if ((!namespace)) {
-            throw new Error('In order to protect other namespaces, namespace param is mandatory');
+
+    async deploy_agents({ image, num_agents, agents_yaml, envs, cpu, mem, pv }) {
+        try {
+            let resources_file_path = path.join(this.output_dir, `${this._namespace}.agents_deployment.${Date.now()}.json`);
+            // modify resources and write to temp yaml
+            const resources = await this.read_resources(agents_yaml);
+            const statefulset = resources.find(res => res.kind === 'StatefulSet');
+            const statefulset_name = statefulset.metadata.name;
```

Comment:
We can write a function that gets the statfulset name and returns it as we use this code i 2 places.